src/Caller.cpp

```diff
@@ -707,17 +645,270 @@ bool Caller::parseOptions(int argc, char **argv) {
+      TabReader reader(hederRow);
+      
+      /* Keep track on column index */
+      column_index = 0;
+      /* Go through all columns to print header and search for missing charge state */
+      while (!reader.error()) {
+        std::string optionalHeader = reader.readString();
+        
+        /* Check if charge state is missing */
+        if (columnMap[column_index] != optionalHeader) {
+          /* Charge state is missing at  'column_index' */
+          missingCols.push_back(column_index);
+          nextColumn=true;
+          if (firstFile) {
+            outFile << columnMap[column_index] << "\t";
+          }
+          /* Check if there are any adjacent missing charge states */
+          while(nextColumn) {
+            column_index++;
+            if (columnMap[column_index] == optionalHeader) {
+              /* No missing charge state, continue */
+              nextColumn=false;
+            } else {
+              /* Missing charge state */
+              missingCols.push_back(column_index);
+              if (firstFile) {
+                outFile << columnMap[column_index] << "\t";
+              }             
+            }
+          }
+        } 
+        if (firstFile) {
+          outFile << columnMap[column_index] << "\t";
+        }
+        column_index++;
+      }
+      if (firstFile) {
+        outFile <<  "\n";
+      }
+      /* Go through rest of the rows in pin-file */
+      std::string nextRow;
+      
+      while(getline(pinFileStream, nextRow)) {
+        
+        TabReader readerRow(nextRow);
+
+        /* Keep track on column for missing charge states */
+        int col = 0;
+        while (!readerRow.error()) {           
+          std::string value = readerRow.readString();
+          /* Check for missing charge state */
+          if(std::find(missingCols.begin(), missingCols.end(), col) != missingCols.end()) {
+            nextColumn=true;
+            outFile << 0 << "\t";
+            col++;
+            /* Check for adjecent missing charge states */
+            while(nextColumn) {
+              if(std::find(missingCols.begin(), missingCols.end(), col) != missingCols.end()) {
+                /* Missing adjecent charge state  */
+                outFile << 0 << "\t";
+                col++;
+              } else {
+                /* Charge states  */
+                nextColumn = false;
+              }
+            }  
+          }
+          outFile << value << "\t";
+          col++;
+        }
+        outFile << "\n";
+      }
+      pinFileStream.close();
+      firstFile = false;      
+    }
+    outFile.close();
   }
 
+
+
+  std::cerr << "All files have been read" << std::endl;
+
+  if (cmd.optionSet("protein-decoy-pattern")) protEstimatorDecoyPrefix = cmd.options["protein-decoy-pattern"];
+
+    if (protEstimatorDecoyPrefix == "auto") {
+      protEstimatorDecoyPrefix = decoy_prefix;
+    }
+
+
+  if (cmd.optionSet("fido-protein") || cmd.optionSet("picked-protein")) {
+
```

Comment:
This code should probeably be moved elsewhere, eventually. However, maybe we save that for later.